### DISCUSSION TITLE
Server Custom Hooks

### DISCUSSION THREAD

**Discussion by dmitriigirskii** [vercel/next.js#31689] [App Router]
Hi!

I read about server components and saw custom hooks for server components
[https://github.com/josephsavona/rfcs/blob/server-components/text/0000-server-components.md#capabilities--constraints-of-server-and-client-components](url)

`Server Hooks/Utilities: Developers may also create custom hooks or utility libraries that are designed for the server. All of the rules for Server Components apply. For example, one use-case for server hooks is to provide helpers for accessing server-side data sources.`

How do I create them in **Next": "12.0.5-canary**

**Comment by leerob** [comment]
Hey! We don't have anything built into Next.js for this yet and haven't started exploring what a solution would look like.
